specs/Menu.md

```diff
@@ -1,731 +1,540 @@
-            <div class="ms-ContextualMenu-linkContent linkContent-133"><span>Item 3 (has submenu)</span><i data-icon-name="ChevronRight" aria-hidden="true" class="ms-ContextualMenu-submenuIcon subMenuIcon-151"></i></div>
-         </button>
-      </li>
-   </ul>
-</div>
+The v0 `Popup` component has an alternative prop, `trigger`, which accepts a React component. This prop simplifies the creation of temporary content by autocontrolling the open/dismiss functionality.
+
+```typescript
+// v0 - shorthand trigger
+<Popup
+  trigger={<Button />}
+  content={...}
+/>
+// v0 - children trigger
+<Popup content={...}>
+  <Button icon={<MoreIcon />} title="Show popup" />
+</Popup>
 ```
 
-### Fast-DNA DOM Structure
-
-```HTML
-<div role="menu" class="contextMenu-3-1-49">
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">Item one</span>
-   </div>
-   <hr tabindex="-1" class="divider-3-1-53">
-   <div tabindex="0" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">Item two</span>
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">Item three</span>
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">SubItem one</span>
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">SubItem Two</span>
-   </div>
-</div>
+### Layout variations
+
+The v7 `ContextualMenu` only has one primary layout which is a vertical list of menu items.
+
+The v0 `Menu` component differs clearly in this that the default layout is a horizontal menu. To achieve the same layout as `ContextualMenu` (and the layout defined in this spec) it's necessary to use the `vertical` prop which is `false` by default.
+
+```typescript
+<Menu items={items} vertical />
 ```
 
-#### Comments
+### Open/Dismiss events
 
-Fast-DNA is one of the only menus that doesn't use `li` elements.
+The v7 `ContextualMenu` is intended to be used as a controlled component. The visibiltiy of the menu is controlled using the `hidden` prop whose value should be React state of the cosumer. A separate `onDismiss` prop can also be used that will be invoked during events where the callout tries to close, i.e. click outside the content.
 
-### MaterialUI
+The v0 `Popup` should be compared here, since the v0 `Menu` does not handle open/dismiss events. `Popup` visibility can be controlled using the `open` prop. `Popup` provides a callback prop `onOpenChange` which can be used both to open and dismiss.
 
-_Note:_ Some long class names removed
+However as mentioned above, the general usage of `Popup` is with an autocontrolled trigger. If a trigger is used with the `open` prop, this disables autocontrolling
```

Comment:
I suggest to rephrase this and say that `Popup` implements autoconrolled pattern that allows to use components as controlled and uncontrolled: https://reactjs.org/docs/uncontrolled-components.html

Reply:
Reworded and fixed the labelling of the examples